src/components/molecules/OakTooltip/OakTooltip.tsx

```diff
@@ -80,33 +83,52 @@ export const OakTooltip = ({
     if (!targetElement) {
       return;
     }
-
+    let ticking = false;
     const updateOverlayStyle = () => {
-      const rect = targetElement.getBoundingClientRect();
+      if (!ticking) {
+        requestAnimationFrame(() => {
+          ticking = false;
+          const rect = targetElement.getBoundingClientRect();
 
-      setOverlayStyle({
-        top: rect.top,
-        left: rect.left,
-        width: rect.width,
-        height: rect.height,
-      });
+          setOverlayStyle({
+            top: rect.top,
+            left: rect.left,
+            width: rect.width,
+            height: rect.height,
+          });
+        });
+        ticking = true;
+      }
     };
 
-    const observer = new ResizeObserver(updateOverlayStyle);
-    observer.observe(targetElement);
-    observer.observe(document.documentElement);
+    // We use an intersection observer to detect when the target element is no longer visible
+    const intersection = new IntersectionObserver((entries) => {
+      setIsIntersecting(entries.every((entry) => entry.isIntersecting));
+    });
+    intersection.observe(targetElement);
+
+    // We use a resize observer to detect when the target element's size changes
+    const resize = new ResizeObserver(updateOverlayStyle);
+    resize.observe(targetElement);
+
+    // Update the overlay position on scroll and resize
+    window.addEventListener("scroll", updateOverlayStyle, true);
```

Comment:
Using capture so that any scroll event on the page should trigger here supporting nesting within multiple scrolling ancestors.